src/ashley/api/views.py

```diff
@@ -0,0 +1,124 @@
+                    )
+                    .exclude(
+                        groups__name=context.get_group_role_name(_FORUM_ROLE_MODERATOR)
+                    )
+                    .order_by(self.ordering)
+                )
+            if role == _FORUM_ROLE_MODERATOR:
+                # Gets all the moderators for the current LTIContext.Moderators have
+                # the primitive role student."""
+                return (
+                    User.objects.filter(
+                        is_active=True,
+                        groups__name=context.get_group_role_name(_FORUM_ROLE_MODERATOR),
+                    )
+                    .filter(
+                        groups__name=context.get_group_role_name(_FORUM_ROLE_STUDENT)
+                    )
+                    .order_by(self.ordering)
+                )
+            if role == _FORUM_ROLE_INSTRUCTOR:
+                # Gets all the instructor for the current LTIContext.
+                return User.objects.filter(
+                    is_active=True,
+                    groups__name=context.get_group_role_name(_FORUM_ROLE_INSTRUCTOR),
+                ).order_by(self.ordering)
+
+        # If no role is specified, filter all the users for the current LTIContext
+        return (
+            User.objects.filter(
+                groups__name__startswith=context.get_group_role_name("")
+            )
+            .order_by(self.ordering)
+            .distinct()
+        )
+
+    # pylint: disable=W0221
+    def update(self, request, pk=None):
+        """Update method is used to promote or revoke moderator role on student users"""
+        user = User.objects.get(pk=pk)
+        context = get_current_lti_session(self.request)
+        group_moderator_name = context.get_group_role_name(_FORUM_ROLE_MODERATOR)
+        group_student_name = context.get_group_role_name(_FORUM_ROLE_STUDENT)
+        try:
+            group_moderator = Group.objects.get(name=group_moderator_name)
+        except Group.DoesNotExist as no_group_moderator:
+            raise PermissionDenied() from no_group_moderator
+
+        # Role asked is student, check user is both student and moderator before removing the group
+        if (
+            self.request.data.get("role") == _FORUM_ROLE_STUDENT
+            and user.groups.filter(name=group_moderator_name)
+            and user.groups.filter(name=group_student_name)
+        ):
+            user.groups.remove(group_moderator)
+            user.save()
+        # Role asked is moderator, check user is student and not yet moderator before adding
+        # the group moderator
+        elif (
+            self.request.data.get("role") == _FORUM_ROLE_MODERATOR
+            and user.groups.filter(name=group_student_name)
+            and user.groups.filter(name=group_moderator_name).count() == 0
+        ):
+            user.groups.add(group_moderator)
+            user.save()
+        else:
+            logger.debug("Query is not expected")
```

Comment:
Which query ? I think this message is not explicit, alone in a log file I will never understand what is the purpose of this message without reading the code.